hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/operations/RegistrationOperation.java

```diff
@@ -16,47 +16,68 @@
 
 package com.hazelcast.spi.impl.eventservice.impl.operations;
 
+import com.hazelcast.internal.util.executor.StripedRunnable;
 import com.hazelcast.nio.ObjectDataInput;
 import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.impl.Versioned;
 import com.hazelcast.spi.impl.SpiDataSerializerHook;
 import com.hazelcast.spi.impl.eventservice.impl.EventServiceImpl;
 import com.hazelcast.spi.impl.eventservice.impl.Registration;
 
 import java.io.IOException;
 
-public class RegistrationOperation extends AbstractRegistrationOperation {
+import static com.hazelcast.internal.cluster.Versions.V5_3;
+
+public class RegistrationOperation extends AbstractRegistrationOperation implements Versioned {
 
     private Registration registration;
-    private boolean response;
+    private int orderKey = -1;
 
     public RegistrationOperation() {
     }
 
-    public RegistrationOperation(Registration registration, int memberListVersion) {
+    public RegistrationOperation(Registration registration, int orderKey, int memberListVersion) {
         super(memberListVersion);
         this.registration = registration;
+        this.orderKey = orderKey;
     }
 
     @Override
     protected void runInternal() {
         EventServiceImpl eventService = (EventServiceImpl) getNodeEngine().getEventService();
-        response = eventService.handleRegistration(registration);
+        eventService.executeEventCallback(new StripedRunnable() {
+            @Override
+            public void run() {
+                eventService.handleLocalRegistration(registration);
+            }
+
+            @Override
+            public int getKey() {
+                return orderKey;
+            }
+        });
     }
 
     @Override
     public Object getResponse() {
-        return response;
+        return true;
```

Comment:
you changed the contract of the operation, is it ok?

Reply:
I can see that the result was ignored but still this is a breaking change

Reply:
In the final version, the responses for `RegistrationOperation` and `DeregistrationOperation` are null as inherited from `Operation`. I checked their all usages and verified that the response is always ignored. Finally, they are implementation classes, so whether the change is breaking only concerns us.

Reply:
> they are implementation classes, so whether the change is breaking only concerns us.
they implement client protocol, and event listeners have standard client compatibility guarantees (in particular this means more that just the previous minor version)